### ISSUE TITLE
Reading doesn't work with Google Pixel XL/Android 8.1

### ISSUE THREAD

**Issue by peterjeschke** [pazaan/600SeriesAndroidUploader#202]
#### Your Environment
* Uploader Version Number: 0.6.1
* Your Android Phone Model Name (e.g. Samsung J1 Ace): Google Pixel XL
* Android Version Number (e.g. use 4.4 for 4.4.4): 8.1
* Network Connection at the time the issue occurred (Wi-Fi, Mobile): Working Wifi, no mobile data.
* (if relevant) Nightscout website version (e.g. 0.8.4 Funnel cake, master / dev branch): Not relevant

#### Brief Explanation of Issue
Reading from the pump doesn't work on my Google Pixel XL, running the latest Android version. It recognizes the CNL and tries to read, but the connection times out. Other OTG devices work (I tried a USB-mouse) and the same OTG-adapter and CNL work with a OP3 running Android 8.0.

Could this be an issue with Android 8.1? Are there any reports with this android version?

#### Steps to Reproduce (for bugs)
1. Start Uploader
2. Plug in CNL
3. Wait for first reading

#### Severity Score
I guess a 3 or 4?

**Comment by Pogman** [comment]
Seems a bit odd that it would be working in 8.0 and not 8.1. There are issues with the notification system in android 8 that need sorting but it does work and communicate with the CNL ok.

**Comment by peterjeschke** [comment]
It might be an issue with Pixel devices in general. I don't have access to
another device running 8.1.

Pogman <notifications@github.com> schrieb am So., 25. Feb. 2018 um
21:42 Uhr:

**Comment by Pogman** [comment]
Can you check if the CNL is shown in the registered devices menu in the uploader? Is there any error shown on the CNL itself?

**Comment by peterjeschke** [comment]
Yes, it is shown as registered device. No, there is no error on the CNL
itself.

It's not my own CNL (I'm just the fascinated friend), but I can provide
more detailed logs on wednesday.

Pogman <notifications@github.com> schrieb am Mo., 26. Feb. 2018 um
13:59 Uhr:

**Comment by Pogman** [comment]
Ok there may be an issue with pairing between the pump and cnl as it does seem to be able to communicate. A timeout will be caused by a non-paired cnl.

**Comment by Ausher2** [comment]
Same issue for me. I deleted the connection between my CNL and pump and reconnected, no change. Shows up as a registered device. OTG cable works with a mouse and CNL is recognized. Connection times out.

**Comment by peterjeschke** [comment]
I deleted all app data and tried connecting the CNL again (to get fresh logs) but it doesn't show up as registered device anymore. So it seems that it can sometimes register as a device but times out most of the time.

One time I got the log message `Could not close connection: Expected to get control character '5' Got '50'`. I don't know whether that could be important?

**Comment by pazaan** [comment]
@peterjeschke Is the pump paired with the CNL? Can you send a BG from the meter to the pump (or manual bolus)? If so, perhaps there is an issue with the OTG cable?

**Comment by Ausher2** [comment]
So I spent WAY too much time on Saturday trying to make this work with my Pixel and I think the problem is within Android 8.1 on the Pixel somewhere. I managed to get the OTG notification back on through the notification settings, cleared cache and data for external media and media settings under System Apps in notifications, etc. When I plug the OTG cable into the phone alone, the USB notification now pops up set to Charging. When I plug the CNL into the OTG cable, nothing happens. When I disconnect the CNL from the OTG cable, the notification pops back up. (Seems backward!) Try to select File Transfer as an option but the blue radial button doesn’t show and then the notification disappears. When I launch the 600 series app, it seems to connect to the CNL but always times out. Here’s the kicker - I was giving up for the day to head somewhere, barely disconnected the meter to test, and lo and behold, all the NS data showed up on my phone! I didn’t have time to mess with it anymore and I haven’t been able to recreate it. Long story short, it was possible somehow but I don’t know what the solution was. CNL is paired with the pump - I can send a reading to the pump from it. The OTH cable works with other devices. Something is wrong in the Pixel and I don’t know what!

**Comment by peterjeschke** [comment]
@pazaan the CNL is paired with the pump and in active use. The OTG cable seems to work, because other USB devices (mouse) work just fine.

**Comment by Pogman** [comment]
Something changed between 8.0 and 8.1 causing a break in the way services and notifications are handled. Android 8+ will be addressed for the 0.7.0 release as there are a number of deprecated api issues to look into.

https://github.com/pazaan/600SeriesAndroidUploader/issues/204

Crash on Android 8.1 as tested using Googles emulator:

```
08-14 17:41:42.858 E: FATAL EXCEPTION: main
    Process: info.nightscout.android, PID: 6977
    android.app.RemoteServiceException: Bad notification for startForeground: java.lang.RuntimeException: invalid channel for service notification: Notification(channel=null pri=0 contentView=null vibrate=null sound=null defaults=0x0 flags=0x40 color=0x00000000 vis=PUBLIC)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1768)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)```

**Comment by jpbillings86** [comment]
Has there been any updates to this? On a Google Pixel 1, Android 9 and get the timeout issue. Yes its paired. Yes it shows up. Yes everything works for other OTG devices. Only this fails.

**Comment by Pogman** [comment]
Fixed for 0.7.0

**Comment by Pogman** [comment]
Closed - completed and available in 0.7.0 release
